In [1]:
import cv2 as cv
import numpy as np
from matplotlib import pyplot as plt

### Reading image

In [3]:
image = cv.imread('Photos/cat.jpg')
cv.imshow('Cat', image)
cv.waitKey(3000)
cv.destroyAllWindows()

### Reading videos

In [5]:
 # 0 for webcam, 1 for first usb cam and so on
capture = cv.VideoCapture('Videos/dog.mp4')
while True:
    isTrue, frame = capture.read()
    cv.imshow('Video', frame)
    if cv.waitKey(20) & 0xFF == ord('d'):
        break
        
capture.release()
cv.destroyAllWindows()

error: OpenCV(4.5.3) C:\Users\runneradmin\AppData\Local\Temp\pip-req-build-sn_xpupm\opencv\modules\imgproc\src\color.cpp:182: error: (-215:Assertion failed) !_src.empty() in function 'cv::cvtColor'


We get 215 assertion failed error when opencv could not find a media file at that particular location that you specified. Now we got the error because the video ran out of frames

### Rescale image

In [6]:
# This works for videos, images, live video
def rescaleFrame(frame, scale = 0.75):
    width = int(frame.shape[1] * scale)
    height = int(frame.shape[0] * scale)
    dimensions = (width, height) 
    return cv.resize(frame, dimensions, interpolation = cv.INTER_AREA)

In [7]:
capture = cv.VideoCapture('Videos/dog.mp4')
while True:
    isTrue, frame = capture.read()
    frameResized = rescaleFrame(frame, 0.5)
    cv.imshow('Rescaled Video', frameResized)
    cv.imshow('Video', frame)
    if cv.waitKey(20) & 0xFF == ord('d'):
        break
        
capture.release()
cv.destroyAllWindows()

In [8]:
# Only works for live video
def changeResolution(width, height):
    capture.set(3, width)
    capture.set(4, height)

### Blank image

In [9]:
# creating a blank image
blank = np.zeros((500, 500, 3), dtype = 'uint8')
cv.imshow('Blank', blank)
cv.waitKey(3000)
cv.destroyAllWindows()

### Paint the image a certain colour

In [10]:
blank[:] = 0,255,0
cv.imshow('Green', blank)
cv.waitKey(3000)
cv.destroyAllWindows()

### Draw a rectangle

In [12]:
# arguments: image, point1, point2, colour, thickness
blank = np.zeros((500, 500, 3), dtype = 'uint8')
cv.rectangle(blank, (0, 0), (250, 250), (255, 255, 255), thickness = 2)
cv.imshow('Rectangle', blank)
cv.waitKey(3000)
cv.destroyAllWindows()

### Draw a circle

In [15]:
# arguments: image, origin, radius in px, colour, thickness
# thickness = cv.FILLED fills the circle or we can use -1
cv.circle(blank, (blank.shape[1] // 2, blank.shape[0] // 2), 40, (0, 0, 255), thickness = cv.FILLED) 
cv.imshow('Circle', blank)
cv.waitKey(3000)
cv.destroyAllWindows()

### Draw a line

In [16]:
# arguments: image, point1, point2, colour, thickness
cv.line(blank, (0, 0), (250, 250), (255, 255, 255), thickness = 2)
cv.imshow('Line', blank)
cv.waitKey(3000)
cv.destroyAllWindows()

### Write text

In [17]:
# arguments: image, text, center, fontname, scale, colour
cv.putText(blank, 'Kotesh', (250, 250), cv.FONT_HERSHEY_COMPLEX, 1.0, (255, 255, 255))
cv.imshow('Text', blank)
cv.waitKey(3000)
cv.destroyAllWindows()

### Convert to grayscale

In [18]:
image = cv.imread('Photos/cat.jpg')
grayscale = cv.cvtColor(image, cv.COLOR_BGR2GRAY)
cv.imshow('Cat', image)
cv.imshow('Gray', grayscale)
cv.waitKey(3000)
cv.destroyAllWindows()

### Blur

In [21]:
image = cv.imread('Photos/cat.jpg')
blur = cv.GaussianBlur(image, (3, 3), cv.BORDER_DEFAULT) # ksize should be odd
cv.imshow('Original', image)
cv.imshow('Blur', blur)
cv.waitKey(3000)
cv.destroyAllWindows()

### Edge cascade

In [38]:
# arguments: image, threshold1, threshold2
image = cv.imread('Photos/park.jpg')
blur = cv.GaussianBlur(image, (7, 7), cv.BORDER_DEFAULT) # ksize should be odd
canny = cv.Canny(blur, 125, 175)
cv.imshow('Park', blur)
cv.imshow('Canny', canny)
cv.waitKey(3000)
cv.destroyAllWindows()

### Dilating the image

In [39]:
dilated = cv.dilate(canny, (7, 7), iterations = 3)
cv.imshow('Canny', canny)
cv.imshow('Dilated', dilated)
cv.waitKey(10000)
cv.destroyAllWindows()

### Eroding

In [41]:
eroded = cv.erode(dilated, (7, 7), iterations = 3)
cv.imshow('Canny', canny)
cv.imshow('Eroded', eroded)
cv.imshow('Dilated', dilated)
cv.waitKey(10000)
cv.destroyAllWindows()

### Resize

In [48]:
image = cv.imread('Photos/group 1.jpg')
cropped1 = cv.resize(image, (600, 600))
cropped2 = cv.resize(image, (600, 600), interpolation = cv.INTER_CUBIC)
cv.imshow('Image', image)
cv.imshow('Cropped Image 1', cropped1)
cv.imshow('Cropped Image 2', cropped2)
cv.waitKey(10000)
cv.destroyAllWindows()

### Cropping

In [50]:
cropped = image[50:150, 200:300]
cv.imshow('Image', image)
cv.imshow('Cropped Image', cropped)
cv.waitKey(10000)
cv.destroyAllWindows()